cabby/model/text/dual_encoder/dataset_run.py

```diff
@@ -0,0 +1,180 @@
+
+    data['cellid'] = data.point.apply(
+      lambda x: gutil.cellid_from_point(x, s2level))
+
+    data['neighbor_cells'] = data.cellid.apply(
+      lambda x: gutil.neighbor_cellid(x))
+
+    data['far_cells'] = data.cellid.apply(
+      lambda cellid: unique_cells_df[unique_cells_df['cellid']==cellid].far.iloc[0])
+
+    cellids_array = np.array(data.cellid.tolist())
+    neighbor_cells_array = np.array(data.neighbor_cells.tolist())
+    far_cells_array = np.array(data.far_cells.tolist())
+
+
+    self.points = data.point.apply(
+      lambda x: gutil.tuple_from_point(x)).tolist()
+    self.labels = data.cellid.apply(lambda x: cellid_to_label[x]).tolist()
+    self.cellids = util.binary_representation(cellids_array, dim = CELLID_DIM)
+    self.neighbor_cells =  util.binary_representation(
+      neighbor_cells_array, dim = CELLID_DIM)
+    self.far_cells =  util.binary_representation(
+      far_cells_array, dim = CELLID_DIM)
+
+    
+  def __getitem__(self, idx: int):
+    '''Supports indexing such that TextGeoDataset[i] can be used to get 
+    i-th sample. 
+    Arguments:
+      idx: The index for which a sample from the dataset will be returned.
+    Returns:
+      A single sample including text, the correct cellid, a neighbor cellid, 
+      a far cellid, a point of the cellid and the label of the cellid.
+    '''
+    text = {key: torch.tensor(val[idx])
+        for key, val in self.encodings.items()}
+    cellid = torch.tensor(self.cellids[idx])
+    neighbor_cells = torch.tensor(self.neighbor_cells[idx])
+    far_cells = torch.tensor(self.far_cells[idx])
+    point = torch.tensor(self.points[idx])
+    label = torch.tensor(self.labels[idx])
+    
+    sample = {'text': text, 'cellid': cellid, 'neighbor_cells': neighbor_cells, 
+      'far_cells': far_cells, 'point': point, 'label': label}
+
+    return sample
+
+  def __len__(self):
+    return len(self.cellids)
+
+
+def create_dataset(
+                  data_dir: str, 
+                  region: str, 
+                  s2level: int, 
+                  infer_only: bool = False
+) -> dataset_item.TextGeoDataset:
+  '''Loads data and creates datasets and train, validate and test sets.
+  Arguments:
+    data_dir: The directory of the data.
+    region: The region of the data.
+    s2level: The s2level of the cells.
+  Returns:
+    The train, validate and test sets and the dictionary of labels to cellids.
+  '''
+  ds = pd.read_json(os.path.join(data_dir, 'dataset.json'))
+  ds['instructions'] = ds.groupby(['id'])['instruction'].transform(lambda x: ' '.join(x))
+
+  ds = ds.drop_duplicates(subset='id', keep="last")
+
+  columns_keep = ds.columns.difference(
+    ['map', 'id', 'instructions', 'end_point', 'start_point'])
+  ds.drop(columns_keep, 1, inplace=True)
+
+  dataset_size = ds.shape[0]
+  logging.info(f"Size of dataset: {ds.shape[0]}")
+
+  train_ds = ds[ds['map']=='map_1']
+  valid_ds = ds[ds['map']=='map_2']
+  test_ds = ds[ds['map']=='map_3']
```

Comment:
IIRC, you did a x-fold scheme with the maps in RUN before. It's fine to do this here, but maybe add a brief comment to mention how this differs from what you did in RUN previously. Also, I believe we were hoping to basically do RUN evaluation w/o using any RUN training data?